packages/component/src/Middleware/ActivityStatus/SendStatus/SendStatus.js

```diff
@@ -0,0 +1,79 @@
+import { Constants } from 'botframework-webchat-core';
+import PropTypes from 'prop-types';
+import React, { useCallback } from 'react';
+
+import connectToWebChat from '../../../connectToWebChat';
+import ScreenReaderText from '../../../ScreenReaderText';
+import SendFailedRetry from './SendFailedRetry';
+import useFocusSendBox from '../../../hooks/useFocusSendBox';
+import useLocalize from '../../../hooks/useLocalize';
+import usePostActivity from '../../../hooks/usePostActivity';
+import useStyleSet from '../../../hooks/useStyleSet';
+
+const {
+  ActivityClientState: { SEND_FAILED, SENDING, SENT }
+} = Constants;
+
+const connectSendStatus = (...selectors) =>
+  connectToWebChat(
+    ({ focusSendBox, language, postActivity }, { activity }) => ({
+      language,
+      retrySend: evt => {
+        evt.preventDefault();
+
+        postActivity(activity);
+
+        // After clicking on "retry", the button will be gone and focus will be lost (back to document.body)
+        // We want to make sure the user stay inside Web Chat
+        focusSendBox();
+      }
+    }),
+    ...selectors
+  );
+
+const SendStatus = ({ activity, sendState }) => {
+  const [{ sendStatus: sendStatusStyleSet }] = useStyleSet();
+  const focusSendBox = useFocusSendBox();
+  const postActivity = usePostActivity();
+
+  // TODO: [P4] Currently, this is the only place which use a templated string
+  //       We could refactor this into a general component if there are more templated strings
```

Comment:
File issue & link

Reply:
I think these comments are for "Send failed. Retry." And they are now Markdown, no need templating. 😉